src/components/Modals/Send/views/Details.tsx

```diff
@@ -46,26 +48,19 @@ export const Details = () => {
   const history = useHistory()
   const translate = useTranslate()
 
-  const {
-    accountId,
-    amountFieldError,
-    asset,
-    cryptoAmount,
-    cryptoSymbol,
-    fiatAmount,
-    fiatSymbol,
-    memo,
-  } = useWatch({
-    control,
-  }) as Partial<SendInput>
+  const { accountId, amountFieldError, assetId, cryptoAmount, fiatAmount, fiatSymbol, memo } =
+    useWatch({
+      control,
+    }) as Partial<SendInput>
 
   const handleAccountChange = useCallback(
     (accountId: AccountId) => {
       setValue(SendFormFields.AccountId, accountId)
-      setValue(SendFormFields.CryptoAmount, '')
-      setValue(SendFormFields.FiatAmount, '')
+      // TODO(gomes): find better heuristics for these - this will rug resetting crypto/fiat amount on AccountId change
+      if (!cryptoAmount) setValue(SendFormFields.CryptoAmount, '')
+      if (!fiatAmount) setValue(SendFormFields.FiatAmount, '')
```

Comment:
@reallybeard FYI - this just doesn't play well with amounts now being set initially if the URL contains an amount, vs. user-set previously